README.adoc

```diff
@@ -36,29 +37,161 @@ include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/
+`src/main/java/hello/PubsubApplication.java`
+[source,java]
+----
+include::/complete/src/main/java/hello/PubsubApplication.java[tag=messageReceiver]
+----
+
+The `ServiceActivator` input channel name must match the method name of the inbound channel declared
+above. In this example, the message body is simply logged and the message is acknowledged. The
+acknowledgement is done through the `AckReplyConsumer` object, which is sent in the message headers.
+
+== Create an outbound channel adapter
+
+An outbound channel adapter listens to new messages in an application's Spring channel and publishes
+them to a Google Cloud Pub/Sub topic.
+
+Instantiating an outbound channel adapter requires a `PubsubTemplate`, which is Spring's abstraction
+to publish messages to Google Cloud Pub/Sub topics. The Spring Cloud GCP Pub/Sub Boot starter
+provides a pre-configured `PubsubTemplate`.
+
+`src/main/java/hello/PubsubApplication.java`
+[source,java]
+----
+include::/complete/src/main/java/hello/PubsubApplication.java[tag=messageSender]
+----
 
+After the outbound channel adapter is instantiated, the name of the Google Cloud Pub/Sub topic where
+messages will be published to must be set. That topic must already exist before the application is
+run. New topics can be created through the https://console.cloud.google.com/[Google Cloud Console]
+or, programatically, with the `PubsubAdmin` class.
 
-`src/main/java/hello/Application.java`
+To write messages to a channel, in order to then publish them to Google Cloud Pub/Sub, a
+`MessagingGateway` can used.
+
+`src/main/java/hello/PubsubApplication.java`
 [source,java]
 ----
-include::complete/src/main/java/hello/Application.java[]
+include::/complete/src/main/java/hello/PubsubApplication.java[tag=messageGateway]
+----
+
+From this code, Spring auto-generates an object that can then be autowired into a private field in
+our application.
+
+`src/main/java/hello/WebAppController.java`
+[source,java]
 ----
+include::/complete/src/main/java/hello/WebAppController.java[tag=autowireGateway]
+----
+
+== Create a resource controller
+
+Create a new controller for your Spring application that lets you write to a Spring channel:
+
+`src/main/java/hello/WebAppController.java`
+[source,java]
+----
+include::complete/src/main/java/hello/WebAppController.java[]
+----
+
+== Make the application executable
+
+Although it is possible to package this service as a traditional link:/understanding/WAR[WAR] file for deployment to an external application server, the simpler approach demonstrated below creates a standalone application. You package everything in a single, executable JAR file, driven by a good old Java `main()` method. Along the way, you use Spring's support for embedding the link:/understanding/Tomcat[Tomcat] servlet container as the HTTP runtime, instead of deploying to an external instance.
```

Comment:
the definition of the main method missing - I think it would be useful to have it here explicitly. 
```
    public static void main(String[] args) throws IOException {
        SpringApplication.run(PubsubApplication.class, args);
    }
```